src/search.js

```diff
@@ -84,11 +85,36 @@ const libHasEmoji = (name, term) => {
     emojilib.lib[name].keywords.some((keyword) => keyword.includes(term))
 }
 const matches = (terms) => {
-  return emojiNames.filter((name) => {
-    return terms.every((term) => {
-      return name.includes(term) || libHasEmoji(name, term)
-    })
-  })
+  const emojiNameResults = [];
+  const emojiKeywordResults = [];
+
+  for (const emojiName of emojiNames) {
+    let hasNameMatch = false;
+    let hasMatch = true;
+
+    for (const term of terms) {
+      if (emojiName.includes(term)) {
+        hasNameMatch = true;
+        continue;
+      } else if (!libHasEmoji(emojiName, term)) {
+        hasMatch = false;
+        break;
+      }
+    }
+
+    if (!hasMatch) {
+      continue;
+    }
+
+    if (hasNameMatch) {
+      emojiNameResults.push(emojiName);
+    } else {
+      emojiKeywordResults.push(emojiName);
+    }
+  }
+
+  // Prioritize emojis that were matched on the name rather than on keywords
+  return [...emojiNameResults, ...emojiKeywordResults]
```

Comment:
Would be great to have more advanced scoring (how many terms match on name VS keywords), but I think it's a good first step.